tests/test_unit/test_arguments.py

```diff
@@ -98,3 +112,226 @@ def test_argument_parameter_build(arguments, expected_built_arguments):
         )._build_arguments()
         == expected_built_arguments
     )
+
+
+model_artifact = ModelArtifact(name="artifact-name", from_="somewhere")
```

Comment:
nit: You've extracted this from one test case, but I'm not sure it increases the readability -- there's other test cases we could have reused it in, and other duplicated artifacts we could extract. Could we be more consistent with one approach or the other?

Reply:
This was created outside of the `parametrize` specifically to test the "don't touch the original" behaviour of `.copy`

Reply:
I think that deserves its own separate test?

Reply:
What do you mean by separate?

Reply:
As in a separate test function that explicitly checks that the input object has not been modified in the 'then' section

Reply:
Makes sense, will do!